File: chaeeun/Programmers/Lv2/짝지어 제거하기.js
// 스택 참고

function solution(s) {
    stack = []
    
    let i = 0
    while (true) {
        if (stack[stack.length-1] != s[i])
            stack.push(s[i])
        else
            stack.pop()
        i++
        if (i == s.length) break
    }
    return stack.length == 0 ? 1 : 0;
}

File: hyesu/SwiftAlgorithm/SwiftAlgorithm/Lv2/짝지어 제거하기.swift
//
//  짝지어 제거하기.swift
//  SwiftAlgorithm
//
//  Created by 김혜수 on 2023/02/24.
//

import Foundation

func solution(_ s: String) -> Int {
    var stack: [Character] = []
    
    for i in s {
        if let last = stack.last,
           last == i {
            stack.removeLast()
        } else {
            stack.append(i)
        }
    }
    
    if stack.isEmpty {
        return 1
    }
        
    return 0
}

print(solution("baabaa"))

